src/overlays/actors/ovl_Bg_Haka_Tomb/z_bg_haka_tomb.c

```diff
@@ -21,26 +30,100 @@ const ActorInit Bg_Haka_Tomb_InitVars = {
     (ActorFunc)BgHakaTomb_Update,
     (ActorFunc)BgHakaTomb_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Haka_Tomb_0x80BD6580/BgHakaTomb_Init.asm")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+Vec3f D_80BD68A4 = { 30.0f, 90.0f, 0.0f };
+
+void BgHakaTomb_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaTomb* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    BcCheck3_BgActorInit(&this->dyna, 1);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000EE8);
+    func_8013E3B8(&this->dyna.actor, &this->unk160, 1);
+    func_80BD6624(this);
+}
+
+void BgHakaTomb_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaTomb* this = THIS;
+
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80BD6624(BgHakaTomb* this) {
+    this->actionFunc = func_80BD66AC;
+}
+
+s32 func_80BD6638(s16* arg0, s16* arg1, s32 arg2) {
+    s32 pad;
+    s32 retVal = false;
+
+    *arg0 = ActorCutscene_GetCurrentIndex();
+    if (*arg0 >= 0 && arg2 > 0) {
+        s32 i = 0;
+        do {
+            if (*arg0 == arg1[i]) {
+                retVal = true;
+                break;
+            }
+            i++;
+
+        } while (i != arg2);
```

Comment:
```suggestion
        s32 i;
        for (i = 0; i != arg2; i++) {
            if (*arg0 == arg1[i]) {
                retVal = true;
                break;
            }
        }
```
Now that this function matches nicely, would this match? Even better, would it match if you put the `s32 i` at the start? Even better, would this still match if you use `for (i = 0; i < arg2; i++) {`

Reply:
Oh wow, I was able to spruce this function up a bunch now you're right